deploy/docker/cp-tools/research/cellprofiler-web-api/app/src/hcs_manager.py

```diff
@@ -153,6 +164,186 @@ def _collect_parent_pipeline_inputs(self, parent_pipeline: HcsPipeline, projecti
             parent_pipeline_inputs.append(parent_pipeline.map_to_file_name(image))
         return parent_pipeline_inputs
 
+    def create_clip(self, params):
+        t = time.time()
+
+        clip_format = params.get('format') if 'format' in params else '.webm'
+        codec = params.get('codec') if 'codec' in params else None
+        fps = int(params.get('fps')) if 'fps' in params else 1
+        sequence_id = params.get('sequenceId') if 'sequenceId' in params else None
+
+        path = self._get_required_field(params, 'path')
+        path = self.extract_local_path(path)
+        by_field = int(self._get_required_field(params, 'byField'))
+        cell = int(self._get_required_field(params, 'cell'))
+
+        preview_dir, sequences = self.parse_hcs(path, sequence_id)
+        preview_dir = self.extract_local_path(preview_dir)
+        index_path = preview_dir + '/index.xml'
+        all_channels = self.get_all_channels(index_path)
+        channels_num = len(all_channels)
+        selected_channel_ids, selected_channels = self.get_selected_channels(params, all_channels)
+
+        clips = []
+        for seq in sequences.keys():
+            timepoints = len(sequences[seq])
+            ome_tiff_path = self.get_ome_tiff_path(preview_dir, seq, by_field)
+            images = imread(ome_tiff_path, key=self.get_pages(selected_channel_ids, channels_num, timepoints, cell))
+            curr = 0
+            for time_point in range(timepoints):
+                channel_images = []
+                for channel in selected_channels:
+                    channel_image = images[curr]
+                    colored_image = self.color_image(channel_image, channel)
+                    channel_images.append(colored_image)
+                    curr = curr + 1
+                merged_image = self.merge_channels(channel_images)
+                clips.append(np.array(merged_image))
+
+        clip_path = CELLPROFILER_API_COMMON_RESULTS_DIR + self.get_clip_file_dir(path)
```

Comment:
Let's use `os.path.join` here (this should be corrected in other places where the path is constructed)